src/sinks/aws_cloudwatch_logs/request.rs

```diff
@@ -98,6 +115,39 @@ impl Future for CloudwatchFuture {
     fn poll(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
         loop {
             match &mut self.state {
+                State::DescribeGroup(fut) => {
+                    let response = match ready!(fut.poll_unpin(cx)) {
+                        Ok(response) => response,
+                        Err(error) => {
+                            return Poll::Ready(Err(CloudwatchError::DescribeLogGroups(error)))
+                        }
+                    };
+
+                    if let Some(group) = response
+                        .log_groups
+                        .ok_or(CloudwatchError::NoGroupsFound)?
+                        .into_iter()
+                        .next()
+                    {
+                        debug!(message = "Group found.", group = ?group.log_group_name);
+                        if self.retention_enabled {
+                            if Some(self.retention_days) != group.retention_in_days {
+                                self.state =
+                                    State::PutRetentionPolicy(self.client.put_retention_policy());
+                                continue;
+                            }
```

Comment:
This seems a bit aggressive to me.  I'd expect to only set the retention policy for _newly created_ log groups, as is the behavior in fluentbit (see `log_retention_days` [here](https://docs.fluentbit.io/manual/pipeline/outputs/cloudwatch)).  As an operator of the system, I want to be sure log groups have _some_ default retention set, but want to leave room for modifying some groups externally without having vector aggressively setting them back.  If this aggressive retention-setting is a desired feature, I think it should be something users should be able to disable.